src/ts/forwarders/proxyAccountFowarder.ts

```diff
@@ -74,227 +76,167 @@ export class ProxyAccountForwarder extends Forwarder<
     );
   }
 
-  /**
-   * Computes the deterministic address for a deployed contract
-   * @param initData Initialisation code for the contract
-   * @param salt One-time use value.
-   */
-  public buildDeployedContractAddress(
-    initData: string,
-    extraData: string
-  ): string {
-    const byteCodeHash = solidityKeccak256(["bytes"], [initData]);
-    const salt = keccak256(extraData);
-
-    const options: Create2Options = {
-      from: this.address,
-      salt: salt,
-      initCodeHash: byteCodeHash,
+  private defaultCallData(
+    data: ProxyAccountCallData
+  ): Required<ProxyAccountCallData> {
+    return {
+      to: data.to,
+      value: data.value ? data.value : 0,
+      data: data.data ? data.data : "0x",
+      callType: data.callType ? data.callType : CallType.CALL,
     };
-
-    return getCreate2Address(options);
-  }
-  /**
-   * Standard encoding for contract call data
-   * @param data The target contract, value (wei) to send, and the calldata to execute in the target contract
-   */
-  protected getEncodedCallData(data: RequiredTo<ProxyAccountCallData>) {
-    return defaultAbiCoder.encode(
-      ["uint", "address", "uint", "bytes"],
-      [
-        data.callType ? data.callType : CallType.CALL,
-        data.to,
-        data.value ? data.value : 0,
-        data.data ? data.data : "0x",
-      ]
-    );
   }
 
-  /**
-   * Returns the encoded calldata for creating a proxy contract
-   * No need for ForwardParams as no signature is required in ProxyAccountDeployer
-   * @returns The proxy deployer address and the calldata for creating proxy account
-   * @throws If the proxy account already exists
-   */
-  public async createProxyContract(): Promise<MinimalTx> {
-    const callData = this.proxyDeployer.interface.functions.createProxyAccount.encode(
-      [await this.signer.getAddress()]
-    );
+  private defaultRevertableCallData(
+    data: RevertableProxyAccountCallData
+  ): Required<RevertableProxyAccountCallData> {
+    return {
+      ...this.defaultCallData(data),
+      revertOnFail: data.revertOnFail ? data.revertOnFail : false,
+    };
+  }
 
-    // 115k gas inc the transaction cost.
+  protected toCallData(
+    initCode: string,
+    extraData: string,
+    value: BigNumberish
+  ): ProxyAccountCallData {
     return {
-      to: this.proxyDeployer.address,
-      data: callData,
+      ...this.encodeForDeploy(initCode, extraData, value),
+      value: "0",
+      callType: CallType.DELEGATE,
     };
   }
 
-  /**
-   * Encodes the forward parameters such that it can be included in
-   * an Ethereum Transaction's data field.
-   * @param params Forward Parameters
-   */
-  protected async encodeSignedMetaTransaction(
-    params: ForwardParams
-  ): Promise<MinimalTx> {
-    const proxyAccount = new ProxyAccountFactory(this.signer).attach(params.to);
+  protected toBatchCallData(
```

Comment:
same comment as before. it is deployment specific right?

Reply:
Fixed